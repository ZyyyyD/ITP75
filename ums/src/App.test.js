import React from 'react';
import { create } from 'react-test-renderer';
import '@testing-library/jest-dom';
import { MemoryRouter } from 'react-router-dom'; // Change to MemoryRouter
import Login from "./components/Home/login";
import { render } from '@testing-library/react';
import Settings from '@mui/icons-material/Settings';


// ... rest of the imports


describe('Login Component', () => {
  it('renders correctly', () => {
    const { container } = render(<MemoryRouter><Login /></MemoryRouter>);
    expect(container).toMatchSnapshot();
  });
  describe('Settings Component', () => {
    it('renders correctly', () => {
      const { container } = render(<MemoryRouter><Settings /></MemoryRouter>);
      expect(container).toMatchSnapshot();
    });
    describe('Users Component Rendering', () => {
      it('renders correctly with default data', () => {
        const { container } = render(
          <MemoryRouter>
            {/* Add any necessary setup for Material-UI */}
            <Users />
          </MemoryRouter>
        );
        expect(container).toMatchSnapshot();
      });
    });
  });
});
