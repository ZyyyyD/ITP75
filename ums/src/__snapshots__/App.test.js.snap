// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component Settings Component renders correctly 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
    data-testid="SettingsIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M19.14 12.94c.04-.3.06-.61.06-.94 0-.32-.02-.64-.07-.94l2.03-1.58c.18-.14.23-.41.12-.61l-1.92-3.32c-.12-.22-.37-.29-.59-.22l-2.39.96c-.5-.38-1.03-.7-1.62-.94l-.36-2.54c-.04-.24-.24-.41-.48-.41h-3.84c-.24 0-.43.17-.47.41l-.36 2.54c-.59.24-1.13.57-1.62.94l-2.39-.96c-.22-.08-.47 0-.59.22L2.74 8.87c-.12.21-.08.47.12.61l2.03 1.58c-.05.3-.09.63-.09.94s.02.64.07.94l-2.03 1.58c-.18.14-.23.41-.12.61l1.92 3.32c.12.22.37.29.59.22l2.39-.96c.5.38 1.03.7 1.62.94l.36 2.54c.05.24.24.41.48.41h3.84c.24 0 .44-.17.47-.41l.36-2.54c.59-.24 1.13-.56 1.62-.94l2.39.96c.22.08.47 0 .59-.22l1.92-3.32c.12-.22.07-.47-.12-.61l-2.01-1.58zM12 15.6c-1.98 0-3.6-1.62-3.6-3.6s1.62-3.6 3.6-3.6 3.6 1.62 3.6 3.6-1.62 3.6-3.6 3.6z"
    />
  </svg>
</div>
`;

exports[`Login Component renders correctly 1`] = `
<div>
  <section
    class="vh-100 gradient-custom"
    data-testid="login-component"
  >
    <div
      class="container h-20"
    >
      <div
        class="row d-flex justify-content-center align-items-center h-100"
      >
        <div
          class="col-12 col-md-8 col-lg-6 col-xl-5"
        >
          <div
            class="card bg-dark text-white"
            style="border-radius: 1rem;"
          >
            <div
              class="card-body p-5 text-center"
            >
              <div
                class=" m-3"
              >
                <img
                  alt=""
                  src="logo.png"
                  style="width: 150px;"
                />
                <h3
                  class="fw-bold pb-3 text-uppercase"
                >
                  User Management System
                </h3>
                <div
                  class="form-outline text-start form-white mb-4"
                >
                  <label
                    class="form-label "
                    for="typeEmailX"
                  >
                    Email Address
                  </label>
                  <input
                    class="form-control form-control-lg"
                    id="typeEmailX"
                    type="email"
                    value=""
                  />
                </div>
                <div
                  class="form-outline text-start  form-white mb-4"
                >
                  <label
                    class="form-label"
                    for="typePasswordX"
                  >
                    Password
                  </label>
                  <input
                    class="form-control form-control-lg"
                    id="typePasswordX"
                    type="password"
                    value=""
                  />
                  <p
                    class="small text-end p-2"
                  >
                    <a
                      class="text-white-50  "
                      href="#!"
                      style="text-decoration: none;"
                    >
                      Forgot password?
                    </a>
                  </p>
                </div>
                <button
                  class="d-block btn btn-outline-light btn-lg "
                  style="margin: auto;"
                >
                  Login
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
